diff --git a/src/FacetManager/DefaultFacetManager.php b/src/FacetManager/DefaultFacetManager.php
index 6fe1a04..a2605da 100644
--- a/src/FacetManager/DefaultFacetManager.php
+++ b/src/FacetManager/DefaultFacetManager.php
@@ -13,6 +13,7 @@ use Drupal\facets\Processor\PreQueryProcessorInterface;
 use Drupal\facets\Processor\ProcessorInterface;
 use Drupal\facets\Processor\ProcessorPluginManager;
 use Drupal\facets\QueryType\QueryTypePluginManager;
+use Drupal\facets\Result\Result;
 use Symfony\Component\BrowserKit\Request;
 use Symfony\Component\HttpFoundation\RequestStack;
 
@@ -370,7 +371,15 @@ class DefaultFacetManager {
             ],
           ],
         ];
-      } else {
+
+      }
+      elseif ($empty_behavior['behavior'] == 'render') {
+        if (empty($facet->getResults())) {
+          $facet->setResults([new Result($facet, '', '', 1)]);
+          $content = $widget->build($facet);
+        }
+       }
+       else {
         // If the facet has no results, but it is being rendered trough ajax we
         // should render a container (that is empty). This is because the
         // javascript needs to be able to find a div to replace with the new
diff --git a/src/Form/FacetForm.php b/src/Form/FacetForm.php
index e35b563..881de3a 100644
--- a/src/Form/FacetForm.php
+++ b/src/Form/FacetForm.php
@@ -385,7 +385,7 @@ class FacetForm extends EntityForm {
       '#type' => 'radios',
       '#title' => $this->t('Empty facet behavior'),
       '#default_value' => $empty_behavior_config['behavior'] ?: 'none',
-      '#options' => ['none' => $this->t('Do not display facet'), 'text' => $this->t('Display text')],
+      '#options' => ['none' => $this->t('Do not display facet'), 'text' => $this->t('Display text'), 'render' => 'Show empty facet'],
       '#description' => $this->t('Take this action if a facet has no items.'),
       '#required' => TRUE,
     ];
