diff --git a/src/Plugin/search_api/backend/SearchApiSolrBackend.php b/src/Plugin/search_api/backend/SearchApiSolrBackend.php
index 2d732e1..bfac13f 100644
--- a/src/Plugin/search_api/backend/SearchApiSolrBackend.php
+++ b/src/Plugin/search_api/backend/SearchApiSolrBackend.php
@@ -1129,16 +1129,16 @@ class SearchApiSolrBackend extends BackendPluginBase {
       // Search API property names. This reverses the mapping in
       // SearchApiSolrBackend::getFieldNames().
       foreach ($field_names as $search_api_property => $solr_property) {
-        if (isset($doc_fields[$solr_property])) {
+        if (isset($doc_fields[$solr_property]) && isset($fields[$search_api_property])) {
+          $field = clone $fields[$search_api_property];
+
           // Date fields need some special treatment to become valid date values
           // (i.e., timestamps) again.
-          if (isset($fields[$search_api_property])
-              && $fields[$search_api_property]->getType() == 'date'
+          if ($field->getType() == 'date'
               && preg_match('/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$/', $doc_fields[$solr_property][0])) {
             $doc_fields[$solr_property][0] = strtotime($doc_fields[$solr_property][0]);
           }
 
-          $field = SearchApiUtility::createField($index, $search_api_property);
           $field->setValues($doc_fields[$solr_property]);
           $result_item->setField($search_api_property, $field);
         }
